mongo/gridfs/gridfs_examples_test.go

```diff
@@ -0,0 +1,225 @@
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo/gridfs"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func ExampleBucket_OpenUploadStream() {
+	var fileContent []byte
+	var bucket *gridfs.Bucket
+
+	// Specify the Metadata option to include a "metadata" field in the files collection document.
+	uploadOpts := options.GridFSUpload().SetMetadata(bson.D{{"metadata tag", "tag"}})
+	uploadStream, err := bucket.OpenUploadStream("filename", uploadOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err = uploadStream.Close(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Use SetWriteDeadline to force a timeout if the upload does not succeed in 2 seconds.
+	if err = uploadStream.SetWriteDeadline(time.Now().Add(2 * time.Second)); err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err = uploadStream.Write(fileContent); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func ExampleBucket_UploadFromStream() {
+	var fileContent []byte
+	var bucket *gridfs.Bucket
+
+	// Use SetWriteDeadline to force a timeout if the upload does not succeed in 2 seconds.
+	if err := bucket.SetWriteDeadline(time.Now().Add(2 * time.Second)); err != nil {
+		log.Fatal(err)
+	}
+	// To make sure the deadline only applies to this upload, set the deadline to an empty time.Time{} after the
+	// operation is done.
+	defer func() {
+		if err := bucket.SetWriteDeadline(time.Time{}); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Specify the Metadata option to include a "metadata" field in the files collection document.
+	uploadOpts := options.GridFSUpload().SetMetadata(bson.D{{"metadata tag", "tag"}})
+	fileID, err := bucket.UploadFromStream("filename", bytes.NewBuffer(fileContent), uploadOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("new file created with ID %s", fileID)
+}
+
+func ExampleBucket_OpenDownloadStream() {
+	var bucket *gridfs.Bucket
+	var fileID primitive.ObjectID
+
+	downloadStream, err := bucket.OpenDownloadStream(fileID)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := downloadStream.Close(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Use SetReadDeadline to force a timeout if the download does not succeed in 2 seconds.
+	if err = downloadStream.SetReadDeadline(time.Now().Add(2 * time.Second)); err != nil {
+		log.Fatal(err)
+	}
+
+	fileBuffer := bytes.NewBuffer(nil)
+	if _, err := io.Copy(fileBuffer, downloadStream); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func ExampleBucket_DownloadToStream() {
+	var bucket *gridfs.Bucket
+	var fileID primitive.ObjectID
+
+	// Use SetReadDeadline to force a timeout if the download does not succeed in 2 seconds.
+	if err := bucket.SetReadDeadline(time.Now().Add(2 * time.Second)); err != nil {
+		log.Fatal(err)
+	}
+	// To make sure the deadline only applies to this download, set the deadline to an empty time.Time{} after the
```

Comment:
Same comment as above for write deadlines

Reply:
See above.